File: java/utility/ResultBody.java
package voxgig.bluefindecryptxp2pesdk.utility;

import voxgig.bluefindecryptxp2pesdk.core.Context;
import voxgig.bluefindecryptxp2pesdk.core.Response;
import voxgig.bluefindecryptxp2pesdk.core.Result;

final class ResultBody {

  private ResultBody() {}

  static Result resultBody(Context ctx) {
    Response response = ctx.response;
    Result result = ctx.result;

    if (result != null) {
      if (response != null && response.jsonFunc != null && response.body != null) {
        result.body = response.jsonFunc.get();
      }
    }

    return result;
  }
}

File: .sdk/tm/php/utility/ResultBody.php
<?php
declare(strict_types=1);

// BluefinDecryptxP2pe SDK utility: result_body

class BluefinDecryptxP2peResultBody
{
    public static function call(BluefinDecryptxP2peContext $ctx): ?BluefinDecryptxP2peResult
    {
        $response = $ctx->response;
        $result = $ctx->result;
        if ($result && $response && $response->json_func && $response->body) {
            $result->body = ($response->json_func)();
        }
        return $result;
    }
}
